code-challenges.test.js

```diff
@@ -14,41 +18,106 @@
+    expect(divisibleByThree(object1)).toEqual("15 is divisible by three")
+    expect(divisibleByThree(object2)).toEqual("0 is divisible by three")
+    expect(divisibleByThree(object3)).toEqual("-7 is not divisible by three")
+  })
+})
+
+// ReferenceError: divisibleByThree is not defined
+
 const object1 = { number: 15 }
-// Expected output: "15 is divisible by three"
+// // Expected output: "15 is divisible by three"
 const object2 = { number: 0 }
-// Expected output: "0 is divisible by three"
+// // Expected output: "0 is divisible by three"
 const object3 = { number: -7 }
-// Expected output: "-7 is not divisible by three"
+// // Expected output: "-7 is not divisible by three"
 
 // b) Create the function that makes the test pass.
 
+// PSUEDO CODE:
+// Create a function with the function name of divisibleByThree
+// Only one parameter (object)
+// Make a conditional statement checking to see if the value is divisble by three
+// Create a return 
+
+const divisibleByThree = (object) => {
+  if (object.number % 3 === 0) {
+    return `${object.number} is divisible by three`
+  } else if (object.number === 0) {
+    return `${object.number} is divisible by three`
+  } else if (object.number % 3 !== 0) {
+    return `${object.number} is not divisible by three`
+  } else {
+    return "Error"
+  }
+}
+
+// Test Suites: 1 passed, 1 total
+
 // --------------------2) Create a function that takes in an array of words and returns an array with all the words capitalized.
 
 // a) Create a test with expect statements for each of the variables provided.
 
+describe("capWords", () => {
+  it("returns an array with all the words capitalized.", () => {
+    expect(capWords(randomNouns1)).toEqual(["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"])
+    expect(capWords(randomNouns2)).toEqual(["Temperature", "Database", "Chopsticks", "Mango"])
+  })
+})
+
+// // ReferenceError: capWords is not defined
+
 const randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
 // Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"]
 const randomNouns2 = ["temperature", "database", "chopsticks", "mango"]
 // Expected output: ["Temperature", "Database", "Chopsticks", "Mango"]
 
 // b) Create the function that makes the test pass.
 
+// PSUEDO CODE:
+// Create a function named capWords
+// Iterate through the array using .map method
+// In order to capitalize each first character I'll need to use charAt(0) which means it's going over only the first character of the index
+// Then I'll use .toUpperCase() to capitalize each first character
+// After I'll need it to stop so I can use .slice(1) which means it will start at the first index character and then using .toLowerCase() ensures everything after the first index is lowercased
+
+function capWords(newArr) {
+  return newArr.map(element => {
+    return element.charAt(0).toUpperCase() + element.slice(1).toLowerCase()
+  })
+}
+
```

Comment:
Hey Zack!  Good work here! Just curious as to why you are using the function keyword instead of ES6 arrow syntax?

Reply:
I was having a hard time doing it the ES6 arrow syntax way, but I knew how to do it the way I did it.